architecture/ambient/lifecycle.md

```diff
@@ -0,0 +1,136 @@
+# Ztunnel Lifecyle
```

Comment:
IMO we should do one of: 

- put this in `ztunnel.md`
- name it `ztunnel-lifecycle.md`
- name it `traffic-capture-lifecycle.md`

Reply:
Also we have:

`TODO: fill in implementation details of how redirection is actually implemented.`

in `ztunnel.md` -> that TODO should probably just link to this doc.

Reply:
Also - we should probably _remove_ some of the text in https://github.com/istio/istio/tree/master/cni/README.MD in favor of this - there's some duplication, and the former is slightly out of date.